### ISSUE TITLE
how to use SparseTensor with ctc loss where the label's length varies within one batch

### ISSUE THREAD

**Issue by ysyyork** [tensorflow/tensorflow#51802]
how to use SparseTensor with ctc loss where the label's length varies within one batch. 

IDK how to formulate a sparse tensor with different lengths and pass into the ctc loss. any example here?

**Comment by mohantym** [comment]
Hi @ysyyork ! Please have a look at this[ link](https://www.tensorflow.org/api_docs/python/tf/nn/ctc_loss) for answer. But We see that the issue template has not been filled, could you please do so as it helps us analyse the issue [tf version, steps followed before you ran into this error or stand alone code/colab gist to reproduce the issue faced].Thanks!

**Comment by ysyyork** [comment]
Hi @mohamedadaly 

Sorry i didn't follow the format because all I want is just some explanation but not a bug. 

But the tf version is 2.4.0. I already read all the API doc about ctc loss but just that I'm a bit confused about how to make it work with SparseTensor. 

My question is that, for example, if I have a labels like below:
```
[
  [1, 0, 1, 0, 1]
  [1, 0, 1, 0, 1, 0, 0, 1, 0, 0]
  [1, 0, 1, 1, 1, 0]
]
```
How to indicate the different lengths with SparseTensor?

For dense labels, I can pad them all to the same length and then pass in the label_length vector to say the lengths are [5, 10, 6]. But if I pass in a SparseTensor, it didn't allow me to pass in the label_length (it has to be None). But to represent the above labels, I can only do 
```
SparseTensor(
    values=[...],
    indices=[...],
    shape=[3, 10]
)
```
If I decode this sparse tensor into a dense one, it becomes the below:
```
[
  [1, 0, 1, 0, 1, 0, 0, 0, 0, 0]
  [1, 0, 1, 0, 1, 0, 0, 1, 0, 0]
  [1, 0, 1, 1, 1, 0, 0, 0, 0, 0]
]
```
Then the first and last labels are actually the same. The length info seems lost in this case. 

So I just want to get  a clearer SparseTensor example about how to formulate this SparseTensor object so that the length info is not lost.

**Comment by mohantym** [comment]
Ok ! We will update the document with an example , Feel free to look at the [comment](https://github.com/tensorflow/tensorflow/issues/41353#issuecomment-658059572) of earlier similar[ issue](https://github.com/tensorflow/tensorflow/issues/41353).  Providing a sample [Gist ](https://colab.research.google.com/gist/mohantym/3a9847de1628a42d12a2d3d64a495d8c/ctc_loss.ipynb#scrollTo=nQX4zZx0eKJG)as an example. Thanks!

**Comment by ysyyork** [comment]
@mohantym Thanks for this. this is helpful. Just to make sure. Does that mean instead of using 1 and 0 as the label, I should use tf.nn.log_softmax(1) and tf.nn.log_softmax(0)?

**Comment by mohantym** [comment]
Hi @ysyyork , Please update once after trying that , You can also look in this [issue ](https://stackoverflow.com/questions/64321779/how-to-use-tf-ctc-loss-with-variable-length-features-and-labels). 
 For any further queries you may open this issue in tf discussion [forum](https://discuss.tensorflow.org/) as there is a larger community there as This is not a bug or feature request,Thank you!

**Comment by ysyyork** [comment]
I have figured out myself. Thanks!

**Comment by mohantym** [comment]
Ok! Could you please close this issue then ?

**Comment by ysyyork** [comment]
sure
